group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -2226,6 +2224,16 @@ private CoordinatorResult<StreamsGroupHeartbeatResult, CoordinatorRecord> stream
             );
         }
 
+        if (group.isShutdownRequested()) {
+            returnedStatus.add(
+                new StreamsGroupHeartbeatResponseData.Status()
+                    .setStatusCode(StreamsGroupHeartbeatResponse.Status.SHUTDOWN_APPLICATION.code())
+                    .setStatusDetail(
+                        "A KafkaStreams instance encountered a fatal error and requested a shutdown for the entire application."
```

Comment:
Is it useful to propagate the member id which request to shutdown entire application?

Reply:
This replicates the message used in the old protocol. But you are right that it would be nice to include the member ID. This requires to change the field from a boolean to an optional string.